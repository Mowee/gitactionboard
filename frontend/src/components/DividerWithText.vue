<template>
  <v-row
    class="my-3"
    align="center"
  >
    <v-col cols="5">
      <v-divider :thickness="thickness" />
    </v-col>
    <v-col class="text-center font-weight-bold">
      OR
    </v-col>
    <v-col cols="5">
      <v-divider :thickness="thickness" />
    </v-col>
  </v-row>
</template>

<script>
export default {
  name: 'DividerWithText',
  props: {
    thickness: {
      type: Number,
      default: 3
    }
  }
};
</script>

<style scoped>

</style>
