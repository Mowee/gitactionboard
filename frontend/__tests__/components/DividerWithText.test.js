import { describe, it, expect } from 'vitest';
import { mountWithWrapper } from '../test-utils';
import DividerWithText from '@/components/DividerWithText';

describe('<DividerWithText />', () => {
  it('should render with default thickness', () => {
    expect(mountWithWrapper(DividerWithText).html()).toMatchSnapshot();
  });

  it('should render with given thickness', () => {
    expect(mountWithWrapper(DividerWithText, {
      props: {
        thickness: 2
      }
    }).html()).toMatchSnapshot();
  });
});
