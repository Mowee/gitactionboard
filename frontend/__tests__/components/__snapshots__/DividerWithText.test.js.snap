// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<DividerWithText /> > should render with default thickness 1`] = `
<div class="v-application v-theme--light v-layout v-layout--full-height v-locale--is-ltr" style="z-index: 1000;">
  <div class="v-application__wrap">
    <div class="v-row align-center my-3">
      <div class="v-col v-col-5">
        <hr aria-orientation="horizontal" class="v-divider v-theme--light" role="separator" style="border-top-width: 3px;">
      </div>
      <div class="v-col text-center font-weight-bold"> OR </div>
      <div class="v-col v-col-5">
        <hr aria-orientation="horizontal" class="v-divider v-theme--light" role="separator" style="border-top-width: 3px;">
      </div>
    </div>
  </div>
</div>
`;

exports[`<DividerWithText /> > should render with given thickness 1`] = `
<div class="v-application v-theme--light v-layout v-layout--full-height v-locale--is-ltr" style="z-index: 1000;">
  <div class="v-application__wrap">
    <div class="v-row align-center my-3">
      <div class="v-col v-col-5">
        <hr aria-orientation="horizontal" class="v-divider v-theme--light" role="separator" style="border-top-width: 2px;">
      </div>
      <div class="v-col text-center font-weight-bold"> OR </div>
      <div class="v-col v-col-5">
        <hr aria-orientation="horizontal" class="v-divider v-theme--light" role="separator" style="border-top-width: 2px;">
      </div>
    </div>
  </div>
</div>
`;
